### PR TITLE
[CSM] Add local node meta reference.

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

File: src/script/lua_api/l_nodemeta_local.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+void NodeMetaLocalRef::reportMetadataChange()
+{
+}
+
+// Exported functions
+
+// garbage collector
+int NodeMetaLocalRef::gc_object(lua_State *L) {
```
Comment (rubenwardy) [inline src/script/lua_api/l_nodemeta_local.cpp]: bad brace

File: src/script/lua_api/l_nodemeta_local.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	return *(NodeMetaLocalRef**)ud;  // unbox pointer
+}
+
+Metadata* NodeMetaLocalRef::getmeta(bool auto_create)
+{
+	return m_meta;
+}
+
+void NodeMetaLocalRef::clearMeta()
+{
```
Comment (rubenwardy) [inline src/script/lua_api/l_nodemeta_local.cpp]: {}

File: src/script/lua_api/l_client.cpp [hunk 3]
```diff
@@ -182,6 +185,16 @@ int ModApiClient::l_get_wielded_item(lua_State *L)
 	return 1;
 }
 
+// get_meta(pos)
+int ModApiClient::l_get_meta(lua_State *L)
+{
+	// Do it
```
Comment (rubenwardy) [inline src/script/lua_api/l_client.cpp]: comment not needed (I realised you probably copy and pasted)

**Comment by red-001** [comment]
Would https://github.com/red-001/minetest/commit/609001ee5731b19566d6e1b26dc93334fa3e1f15 be better?

File: src/script/lua_api/l_nodemeta_local.cpp (hunk 2 above)
Comment (nerzhul) [inline src/script/lua_api/l_nodemeta_local.cpp]: i'm not sure clang-format like it, it's not very important. Also if it's really empty just keep it in header file

**Comment by red-001** [comment]
@rubenwardy done see #5508

**Pull request closed without merging**